### PR TITLE
improved flag system and subscription feature

### PR DESCRIPTION
**Pull request by juanCortelezzi**
Hey there,
I just added the "subscription" feature alongside the main functionality. 
i have tested it for an hour now and i think its a great addition.

### CODE REVIEW

File: yt [hunk 1]
```diff
@@ -1,69 +1,191 @@
...
+        esac
+    done
 fi
+
+# if no query is set with flags then ask for one
+if [ -z "$query" ]; then
+    # ask for a channel
+    if [[ $action = "c" ]]; then
+        # if no channels in sub list -> exit
+        if [ -z "$query" ]; then
```
Comment (sayan01) [inline yt]: I'm just glancing over the code, but should'nt this be $sublist instead?
```suggestion
        if [ -z "$sublist" ]; then
```

File: yt [hunk 2]
```diff
@@ -1,69 +1,191 @@
...
+fi
+
+# available flags
+optstring=":s:c:gh"
 
 defcmd="fzf"
-guicmd="rofi -dmenu -i" #uncomment next line for dmenu
-#guicmd="dmenu -i -l 15"
+defaction="c"
+guicmd="rofi -dmenu -i"
```
Comment (sayan01) [inline yt]: Please add the dmenu lines back, people using dmenu might also use the script
```suggestion
guicmd="rofi -dmenu -i" #uncomment next line for dmenu
#guicmd="dmenu -i -l 15"
```

File: yt [hunk 3]
```diff
@@ -1,69 +1,191 @@
 #!/bin/bash
-# dependencies: mpv youtube-dl fzf rofi/dmenu
-# search videos and playlists on youtube and play them in mpv, without an API 
-# usage:
-# yt					asks for input in stdin, prompts using fzf
-# yt search query		takes input from the passed arg, prompts using fzf
-# yt -r					takes input and prompts using rofi ($guicmd)
+
```
Comment (sayan01) [inline yt:3]:
```suggestion
# dependencies: mpv youtube-dl fzf rofi/dmenu

```

**Review by sayan01 — CHANGES_REQUESTED** [review]
Hey Juan! I have glanced over your code, and made some change requests, please see them.
I will test the code properly a little later and merge it then.

File: yt (hunk 2 above)
Comment (juanCortelezzi) [reply yt]: Sorry, I forgot to add it back.

File: yt (hunk 1 above)
Comment (juanCortelezzi) [reply yt]: Yes, hahahaha i copied it from above and forgot to change it.

**Review by juanCortelezzi — COMMENTED** [review]
Your comments are correct, feel free to make the changes and do some tests. Thanks for your time !

File: yt (hunk 1 above)
Comment (sayan01) [reply yt]: yeah i guessed so 😄
no worries, i'll change that

**Comment by sayan01** [comment]
@juanCortelezzi I have added some more features to your PR, read the commit (c5002cf17c8167d34d48e224b7aebef92a9b0bd3) for details. Thanks for this contribution!

**Comment by juanCortelezzi** [comment]
Perfect, thank you very much !

**Pull request merged**